perf/sawtooth_perf/src/batch_gen.rs

```diff
@@ -130,36 +130,70 @@ impl<'a> SignedBatchProducer<'a> {
             signer: signer,
         }
     }
+}
+
+impl<'a> BatchIter for SignedBatchProducer<'a> {
 
     /// Gets the next BatchResult.
     /// `Ok(None)` indicates that the underlying source has been consumed.
-    pub fn next_batch(&mut self) -> BatchResult {
+    fn next_batch(&mut self) -> BatchResult {
         let txns = match self.transaction_source.next(self.max_batch_size) {
             Ok(txns) => txns,
             Err(err) => return Err(BatchingError::MessageError(err)),
         };
 
-        if txns.len() == 0 {
-            return Ok(None);
-        }
+        batch_transactions(txns, self.signer)
+    }
+}
+
+fn batch_transactions(txns: Vec<Transaction>, signer: &signing::Signer) -> BatchResult {
+    if txns.len() == 0 {
+        return Ok(None);
+    }
+
+    let mut batch_header = BatchHeader::new();
+
+    // set signer_public_key
+    let txn_ids = txns.iter().cloned().map(|mut txn| txn.take_header_signature()).collect();
+    batch_header.set_transaction_ids(protobuf::RepeatedField::from_vec(txn_ids));
+    batch_header.set_signer_public_key(signer.get_public_key()?.as_hex());
+
+    let header_bytes = batch_header.write_to_bytes().unwrap();
+    let signature = try!(signer.sign(&header_bytes).map_err(BatchingError::SigningError));
 
-        let mut batch_header = BatchHeader::new();
+    let mut batch = Batch::new();
+    batch.set_header_signature(signature);
+    batch.set_header(header_bytes);
+    batch.set_transactions(protobuf::RepeatedField::from_vec(txns));
 
-        // set signer_public_key
-        let txn_ids = txns.iter().cloned().map(|mut txn| txn.take_header_signature()).collect();
-        batch_header.set_transaction_ids(protobuf::RepeatedField::from_vec(txn_ids));
-        batch_header.set_signer_public_key(self.signer.get_public_key()?.as_hex());
+    Ok(Some(batch))
+}
+
+pub trait BatchIter {
```

Comment:
This should just be the `std::iter::Iterator`

Reply:
Sounds good.